server/src/main/java/org/opensearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -216,8 +227,12 @@ public BytesRef indexedValueForSearch(Object value) {
             switch (sValue) {
                 case "true":
                     return Values.TRUE;
+                case "T":
```

Comment:
@harshavamsi could we please extract this change to separate pull request? I the nutshell - we are changing the existing behavior with the potential to break existing applications.

Reply:
@reta sorry just getting to this now. Do you think this requires a separate change? We're simply adding more cases to determine the inputs. This [change](https://github.com/opensearch-project/OpenSearch/pull/11650/files/edff7d44b921a2d09675e3d63b2679884f4e9dd8#diff-bc4ce3e8726bcd1d3d99ea4b5b654fbb95dbaf59435cb283f909f7d41680f931R211) requires that we add this case.

Reply:
@harshavamsi I think the change by itself makes sense, but it changes the semantic of the boolean field - this could break existing applications, isolating (and not backporting to 2.x) could be less disruptive

Reply:
@harshavamsi - Please correct me if I am wrong, but this should break existing applications, if we are removing the set of possible values. Since, we are including more values the change should be backward compatible!

Reply:
thanks @reta and @jainankitk, after some thought I will revert this change, in fact I plan on changing some aspects of the way the boolean field mapper changes are made. I don't want to use the `IndexOrDocValuesQuery` any more since I don't think it makes sense to use that for boolean values that are converted to 0's and 1's internally. I will simply make a change to allow for searching to happen when we have only doc_values enabled.